### PR TITLE
Fix MutList.toArray

### LINKED ISSUE #1796: MutList.toArray reverses list
```
def main(_args: Array[String]): Int32 & Impure =
    let buf = MutList.new();
    MutList.push!(1, buf);
    MutList.push!(2, buf);
    MutList.push!(3, buf);
    println(buf);
    println(MutList.toArray(buf));
    0
```
Actual output:
```
MutList#{1, 2, 3}
[3, 2, 1]
```
Expected output:
```
MutList#{1, 2, 3}
[1, 2, 3]
```

### PR DESCRIPTION
**Pull request by bdahse**
Fixes #1796 .

### CODE REVIEW

File: main/src/library/MutList.flix [hunk 1]
```diff
@@ -663,14 +663,9 @@ namespace MutList {
...
-        if (l == 0)
-            []
-        else {
-            let a = Array.new(get(0, v), l);
-            foreachWithIndex((x, i) -> a[l - i - 1] = x, v);
-            a
-        }
+        let MutList(a, _) = v;
+        shrink!(v);
+        deref a
```
Comment (magnus-madsen) [inline main/src/library/MutList.flix:668]: Does this not return the same arrays as the one inside MutList? Thus breaking its abstraction. We should always return a copy. I think `Copy.copy` can be used or slice it from beginning to end.

File: main/src/library/MutList.flix [hunk 2]
```diff
@@ -663,14 +663,9 @@ namespace MutList {
...
 
     ///
     /// Returns `true` if the mutable lists `v1` and `v2` have the same elements, i.e. are structurally equal.
```
Comment (magnus-madsen) [inline main/src/library/MutList.flix:671]: Also, I expected some test cases to ensure that the new implementation is correct.

File: main/src/library/MutList.flix (hunk 1 above)
Comment (bdahse) [reply main/src/library/MutList.flix:668]: Yes when length == capacity we return the same array. Oops.

**Comment by magnus-madsen** [comment]
Thanks! I shall rebase the Datalog PR onto this in a bit. That way you can rebuild your JAR with this change.

**Pull request merged**